cli/src/diff_util.rs

```diff
@@ -520,8 +520,21 @@ impl ColorWordsDiffOptions {
                 })?),
             }
         };
+        let context = if let Some(c) = args.context {
+            c
+        } else {
+            let key = "diff.context";
+            let value = config.get_int(key)?;
+            if value < 0 {
+                return Err(config::ConfigError::Message(format!(
+                    "context must be at least 0, got {value}"
+                )));
+            }
+            usize::try_from(value)
+                .map_err(|err| config::ConfigError::Message(format!("invalid {key}: {err}")))?
```

Comment:
nit: `config.get(key)?` (i.e. `config.get::<usize>()`) should produce a good-enough error message.